pkg/util/kubernetes/apiserver/hpa_controller.go

```diff
@@ -225,30 +231,39 @@ func (h *AutoscalersController) processNext() bool {
 		h.autoscalers <- key
```

Comment:
I recommend moving this to after `syncAutoscalers`

```
err := h.syncAutoscalers(key)
if err != nil {
	log.Errorf("Could not sync HPAs %s", err)
}

if h.autoscalers != nil {
	h.autoscalers <- key
}
```

Reply:
I'll just move the shallow copy inside the lock